examples/gallery/embellishments/scalebar.py

```diff
@@ -0,0 +1,141 @@
+   - **x**: Give plot coordinates as *x*\/\ *y*.
+
+ - length: **+w**. Give a distance value, and, optionally a distance unit.
+   Choose from **e** (meters), **f** (feet), **k** (kilometers) [Default],
+   **M** (statute miles), **n** (nautical miles), or **u** (US survey feet).
+ - origin: **+c**\ [*slon*/]\ *slat*. Control where on the map the scale bar
+   applies. If **+c** is not given the reference point is used. If only
+   **+c** is appended the middle of the map is used. Note that *slon* is only
+   optional for projections with constant scale along parallels, e.g.,
+   Mercator projection.
+ - justify: **+j**. Set the anchor point. Specify a two-character (order
+   independent) code. Choose from vertical **T**\(op), **M**\(iddle), or
+   **B**\(ottom) and horizontal **L**\(eft), **C**\(entre), or **R**\(ight).
+ - offset: **+o**\ *offset* or **+o**\ *xoffset*/\ *yoffset*. Give either a
+   common shift or individual shifts in x (longitude) and y (latitude)
+   directions.
+ - height: Use :gmt-term:`MAP_SCALE_HEIGHT` via :func:`pygmt.config`.
+ - fancy style: **+f**. Get a scale bar that looks like train tracks.
+ - unit: **+u**. Add the distance unit given via **+w** to the single
+   distance values.
+ - label: **+l**. Add the distance unit given via **+w** as label. Append
+   text to get a customized label instead.
+ - alignment: **+a**. Set the label alignment. Choose from **t**\(op)
+   [Default], **b**\(ottom), **l**\(eft), or **r**\(ight).
+"""
+
+# %%
+import pygmt
+
+# Create a new Figure instance
+fig = pygmt.Figure()
+
+# -----------------------------------------------------------------------------
+# Add a plain scale bar
+fig.basemap(
+    region=[-45, -25, -15, 0],
+    projection="M0/0/10c",  # Mercator projection with 10 centimeters width
+    frame=["WSne", "af"],
+    # The scale bar is placed at position (j) MiddleCenter, applies at the
+    # reference point (+c is not given), and represents a length (+w) of 1000
+    # kilometers
+    map_scale="jMC+w1000k",
+)
+
+fig.shift_origin(xshift="+w1c")
+
+# -----------------------------------------------------------------------------
+# Add a fancy scale bar
+fig.basemap(
+    region=[-45, -25, -15, 0],
+    projection="M10c",
+    frame=["wSnE", "af"],
+    # Place the scale bar at position (j) MiddleLeft by using MiddleLeft as
+    # anchor point (+j) with an offset (+o) of 1 centimeter in x direction
+    # (longitude) and 0 centimeters in y direction (latitude)
+    # Use a fancy style (+f) to get a scale bar that looks like train tracks
+    # Add the distance unit (+u) to the single distance values
+    map_scale="jML+jML+o1c/0c+w1000k+f+u",
+)
+
+fig.show()
+
+# %%
+
+# Create a new Figure instance
+fig = pygmt.Figure()
+
+# -----------------------------------------------------------------------------
+# Add a thick scale bar
+# Adjust the GMT default parameter MAP_SCALE_HEIGHT locally (the change applies
+# only to the code within the "with" block)
```

Comment:
```suggestion
# only to the code within the "with" statement)
```

Reply:
Hm. I am still not 100 % sure whether `within the "with" statement` sounds better than `within the "with" block` and makes sense. Maybe we can write `under the "with" statement` if we want to use `statement` instead of `block`?

Reply:
I feel 'within the "with" statement' is better.

Reply:
OK. Should I also update the formulations in the `Figure.inset` line example: 
https://github.com/GenericMappingTools/pygmt/blob/06ae8180fd0b398136e7cfd91932ec98407bff84/pygmt/src/inset.py#L119-L131